gap/main/setup.gi

```diff
@@ -632,21 +635,23 @@ InstallMethod(FakeOne, "for a Rees 0-matrix semigroup element collection",
 InstallMethod(ChooseHashFunction, "for a Rees 0-matrix semigroup element",
 [IsReesZeroMatrixSemigroupElement, IsInt],
 function(x, hashlen)
-  local R, under, func;
+  local R, data, under, func;
 
   R := ReesMatrixSemigroupOfFamily(FamilyObj(x));
   if IsMultiplicativeZero(R, x) then
     x := Representative(UnderlyingSemigroup(R));
     under := ChooseHashFunction(x, hashlen).func;
+    data := ChooseHashFunction(x, hashlen).data;
   else
     under := ChooseHashFunction(x![2], hashlen).func;
+    data := ChooseHashFunction(x![2], hashlen).data;
   fi;
   func := function(x, hashlen)
-    return SEMIGROUPS.HashFunctionRZMSE(x, hashlen, under);
+    return SEMIGROUPS.HashFunctionRZMSE(x, data, under);
   end;
 
   return rec(func := func,
-             data := hashlen);
+             data := data);
 end);
 
```

Comment:
Should probably check that `x` is a perm or pc group element here too, and return an error if it is not.